drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -477,11 +477,13 @@ def do_attempt(self, data, suffix=''):
             misplaced_items.append(self._get_item_definition(int(item_id)))
 
         feedback_msgs = [FeedbackMessage(item['feedback']['incorrect'], None) for item in misplaced_items]
-
+        grade = self._get_raw_earned_if_set()
+        if grade is not None:
+            grade = self.weighted_grade()
```

Comment:
And the idea with this code is to first check if there is a grade at all, then return None if there is no grade, but `weighted_grade()` if there is. A helper function will definitely clean this up.